pkg/cloudprovider/providers/gce/gce.go

```diff
@@ -760,20 +766,29 @@ func (gce *GCECloud) forwardingRuleNeedsUpdate(name, region string, loadBalancer
 		if isHTTPErrorCode(err, http.StatusNotFound) {
 			return false, true, "", nil
 		}
-		return false, false, "", fmt.Errorf("error getting load balancer's forwarding rule: %v", err)
+		// Err on the side of caution in case of errors. Caller should notice the error and retry.
+		// We never want to end up recreating resources because gce api flaked.
+		return true, false, "", fmt.Errorf("error getting load balancer's forwarding rule: %v", err)
 	}
-	if loadBalancerIP != fwd.IPAddress {
+	// TODO: we report loadbalancer IP through status, so we want to verify if
+	// that matches the forwarding rule as well.
+	if loadBalancerIP != "" && loadBalancerIP != fwd.IPAddress {
+		glog.Warningf("Loadbalancer ip for forwarding rule %v changed from %v to %v", fwd.Name, fwd.IPAddress, loadBalancerIP)
 		return true, true, fwd.IPAddress, nil
 	}
 	portRange, err := loadBalancerPortRange(ports)
 	if err != nil {
-		return false, false, "", err
+		// Err on the side of caution in case of errors. Caller should notice the error and retry.
+		// We never want to end up recreating resources because gce api flaked.
+		return true, false, "", err
 	}
 	if portRange != fwd.PortRange {
+		glog.Warningf("Loadbalancer port range for forwarding rule %v changed from %v to %v", fwd.Name, fwd.PortRange, portRange)
```

Comment:
same comment.

it's unlcear whether these cases are really warnings, though.  They seem more like expected conditions when changing a Service..